linux/deploy_vm/ubuntu/get_mainline_kernel.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Set the base url for kernel version check"
+  ansible.builtin.set_fact:
+    mainline_kernel_url: "https://kernel.ubuntu.com/mainline"
+
```

Comment:
Its better to add a assert task to check user's setting about `upgrade_kernel_version`, it should be either `latest` or a valid version. Since a version is always carried with a 'v' prefix, I suggest you making the rule that user need to input version like`vX.Y.Z` not `X.Y.Z` etc